documentation/blog/2023-08-31-view-transitions.md

```diff
@@ -0,0 +1,264 @@
+
+While it’s only implemented in Chrome (and other Chromium-based browsers) for now, [WebKit is also in favor](https://github.com/WebKit/standards-positions/issues/48#issuecomment-1679760489) of it. Even if you’re on an unsupported browser, it’s a perfect candidate for progressive enhancement since we can always fall back to a non-animated navigation.
+
+It’s important to note that view transitions is a browser API, not a SvelteKit one. `onNavigate` is the only SvelteKit-specific API we’ll use today. Everything else can be used wherever you write for the web! For more on the view transitions API, I highly recommend the [Chrome explainer](https://developer.chrome.com/docs/web-platform/view-transitions/) by Jake Archibald.
+
+## How `onNavigate` works
+
+Before learning how to write view transitions, let's highlight the function that makes it all possible: [`onNavigate`](https://kit.svelte.dev/docs/modules#$app-navigation-onnavigate).
+
+Until recently, SvelteKit had two navigation lifecycle functions: [`beforeNavigate`](https://kit.svelte.dev/docs/modules#$app-navigation-beforenavigate), which fires before a navigation starts, and [`afterNavigate`](https://kit.svelte.dev/docs/modules#$app-navigation-afternavigate), which fires after the page has been updated following a navigation. SvelteKit 1.24 introduces a third: `onNavigate`, which will fire on every navigation, immediately before the new page is rendered. Importantly, it will run _after_ any data loading for the page has completed – since starting a view transition prevents any interaction with the page, we want to start it as late as possible.
+
+You can also return a promise from `onNavigate`, which will suspend the navigation until it resolves. This will let us wait to complete the navigation until the view transition has started.
+
+```js
+// @errors: 2304 7006
+function delayNavigation() {
+	return new Promise((res) => setTimeout(res, 100));
+}
+
+onNavigate(async (navigation) => {
+	// do some work immediately before the navigation completes
+
+	// optionally return a promise to delay navigation until it resolves
+	return delayNavigation();
+});
+```
+
+With that out of the way, let's see how you can use view transitions in your SvelteKit app.
+
+## Getting started with view transitions
+
+The best way to see view transitions in action is to try it yourself. You can spin up the SvelteKit demo app by running `npm create svelte@latest` in your local terminal, or in your browser on [StackBlitz](https://sveltekit.new). Make sure to use a browser that supports the view transitions API. Once you have the app running, add the following to the script block in `src/routes/+layout.svelte`.
+
+```js
+// @errors: 2305 7006 2339 2810
+import { onNavigate } from '$app/navigation';
```

Comment:
not a dealbreaker, but I couldn't figure out how to get the right types on hover here
<img width="567" alt="image" src="https://github.com/sveltejs/svelte/assets/4992896/660bfa31-ef45-49e2-a3c5-e2a90042d327">